File: src/main/java/Command.java
import java.io.BufferedReader;
import java.io.File;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.stream.Collectors;

public class Command {
    private final String[] command;
    private String output;

    public static Command of(String command) {
        return new Command(command);
    }

    private Command(String command) {
        this.command = String.format("cmd /c \"%s\"", command).split("\\s");
    }

    void run(String workingDir) throws InterruptedException, IOException {
        Process proc = new ProcessBuilder(this.command)
                .directory(new File(workingDir))
                .redirectOutput(ProcessBuilder.Redirect.PIPE)
                .redirectError(ProcessBuilder.Redirect.PIPE)
                .start();
        proc.waitFor();
        output = new BufferedReader(new InputStreamReader(proc.getInputStream())).lines().collect(Collectors.joining());
    }
}

File: src/main/kotlin/node/command.kt
package node

import java.io.File

fun String.run(workingDir: File): Process {
    val proc = ProcessBuilder(*("cmd /c \"${this}\"".split("\\s".toRegex())).toTypedArray())
        .directory(workingDir)
        .inheritIO()
        .start()
    proc.waitFor()
    return proc
}

fun String.runCommand(workingDir: File): Process {
    val proc = ProcessBuilder(*("cmd /c \"${this}\"".split("\\s".toRegex())).toTypedArray())
        .directory(workingDir)
        .redirectOutput(ProcessBuilder.Redirect.PIPE)
        .redirectError(ProcessBuilder.Redirect.PIPE)
        .start()
    proc.waitFor()
    return proc
}

fun Process.output(): String {
    return this.inputStream.bufferedReader().readText()
}
